ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["intermediate_size"],
+#     )
+#
+#     initializer_range: float = schema_utils.NonNegativeFloat(
+#         default=0.02,
+#      description="The standard deviation of the truncated_normal_initializer for initializing all weight matrices.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["initializer_range"],
+#     )
+#
+#     layer_norm_eps: float = schema_utils.NonNegativeFloat(
+#         default=1e-12,
+#         description="The epsilon used by the layer normalization layers.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["layer_norm_eps"],
+#     )
+#
+#     gradient_checkpointing: bool = schema_utils.Boolean(
+#         default=False,
+#         description="",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["gradient_checkpointing"],
+#     )
+#
+#     patch_size: int = schema_utils.PositiveInteger(
+#         default=16,
+#         description="The image patch size. Each patch is patch_size² pixels. Must evenly divide the image width and "
+#         "height.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["patch_size"],
+#     )
+#
+#     saved_weights_in_checkpoint: bool = schema_utils.Boolean(
+#         default=False,
+#         description="Are the pretrained encoder weights saved in this model's checkpoint? Automatically set to"
+#         "True for trained models to prevent loading pretrained encoder weights from model hub.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["saved_weights_in_checkpoint"],
+#     )
+#
+#     trainable: bool = schema_utils.Boolean(
+#         default=True,
+#         description="Is the encoder trainable.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["trainable"],
+#     )
+#
+#     use_pretrained: bool = schema_utils.Boolean(
+#         default=True,
+#         description="Use pre-trained model weights from Hugging Face.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["use_pretrained"],
+#     )
+#
+#     pretrained_model: str = schema_utils.String(
+#         default="google/vit-base-patch16-224",
+#         description="The name of the pre-trained model to use.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["pretrained_model"],
+#     )
+
+
+@dataclass
+class TVBaseEncoderConfig(BaseEncoderConfig):
+    use_pretrained: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Download model weights from pre-trained model.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["use_pretrained"],
     )
 
-    batch_norm_epsilon: float = schema_utils.NonNegativeFloat(
-        default=0.001,
-        description="Epsilon of the batch norm.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["batch_norm_epsilon"],
+    model_cache_dir: str = schema_utils.String(
```

Comment:
By default, `String` allows `None` as a valid value. If this is intended, please set the type annotation to `Optional`.

Reply:
Got it....added Optional to annotation.